### PR TITLE
Profile page and Edit page

### PR DESCRIPTION
**Pull request by ProgrammingJoe**

### CODE REVIEW

File: public/app/controllers/userController.js [hunk 1]
```diff
@@ -60,7 +60,7 @@ userControllers.controller('userEditController', function($http, User, $routePar
   });
 
   vm.updatedUser = {
-    username: ""
+    username: "",
```
Comment (cpjk) [inline public/app/controllers/userController.js:63]: This seems like a mistake.. :smile:

File: public/app/views/partials/users/show.jade [hunk 2]
```diff
@@ -1,16 +1,26 @@
-div.text-center
+link(rel='stylesheet', href='/stylesheets/style.css')
```
Comment (cpjk) [inline public/app/views/partials/users/show.jade]: this stylesheet is already included in layout.jade

File: public/app/controllers/userController.js (hunk 1 above)
Comment (ProgrammingJoe) [inline public/app/controllers/userController.js:63]: Hmm, yes I think so too. I was fiddling with that and just forgot to remove the comma.

File: public/app/views/partials/users/show.jade (hunk 2 above)
Comment (ProgrammingJoe) [inline public/app/views/partials/users/show.jade]: Ohhh, I understand now.

**Comment by cpjk** [comment]
The first name, last name, and email are not updated on form submission. You added form inputs, but there is nothing to handle the form submission. You need to change the controller action and the api route that handles the submission in order to actually persist the changes to the database.

File: app/routes/api/users.js [hunk 3]
```diff
@@ -92,8 +92,19 @@ router.post('/:username', function(req, res, next){
         res.status(404).send();
       }
       else{
-        if(req.body.username) {
-          user.username = req.body.username;
+        if(req.body.username || req.body.firstname || req.body.lastname || req.body.email) {
```
Comment (cpjk) [inline app/routes/api/users.js]: We don't need this if statement.

File: public/app/views/partials/users/edit.jade [hunk 4]
```diff
@@ -1,10 +1,17 @@
 div.page-header.text-center
-  h3 Editing {{userCtrl.originalUser.username}}
+  h3 Editing User {{userCtrl.originalUser.username}}
+  p.flat-para Update your user information below, fields left blank will not be changed
 
 div.text-center
   form(ng-submit="userCtrl.submit()").centred-form
     div.form-group
-      input(type="text" required placeholder="Username" ng-model="userCtrl.updatedUser.username").form-control
+      input(type="text" placeholder="{{userCtrl.originalUser.username}}" ng-model="userCtrl.updatedUser.username" maxlength="30" value="{{userCtrl.originalUser.username}}").form-control
```
Comment (cpjk) [inline public/app/views/partials/users/edit.jade]: I don't think we need the value on the inputs anymore

File: app/routes/api/users.js (hunk 3 above)
Comment (ProgrammingJoe) [inline app/routes/api/users.js]: I can get rid of mine and the past one right? So there isn't one at all.

Comment (cpjk) [inline app/routes/api/users.js]: Yeah, just get rid of line 95 in the current version. If you look at the logic it is not adding anything.

File: app/routes/api/users.js [hunk 5]
```diff
@@ -92,9 +92,18 @@ router.post('/:username', function(req, res, next){
         res.status(404).send();
       }
       else{
-        if(req.body.username) {
-          user.username = req.body.username;
-        }
+          if(req.body.username != ''){
```
Comment (cpjk) [inline app/routes/api/users.js]: Please indent with only 2 spaces for consistency.

File: app/routes/api/users.js [hunk 6]
```diff
@@ -92,9 +92,18 @@ router.post('/:username', function(req, res, next){
...
+            user.username = req.body.username;
+          }
+          if(req.body.firstname != ''){
```
Comment (cpjk) [inline app/routes/api/users.js]: It also would probably be better to use if(!req.body.firstname) {}

this covers more cases like if firstname is null

**Pull request merged**